synapse/crypto/context_factory.py

```diff
@@ -48,3 +51,45 @@ def configure_context(context, config):
 
     def getContext(self):
         return self._context
+
+
+@implementer(IOpenSSLClientConnectionCreator)
+class ClientTLSOptions(object):
+    """
+    Client creator for TLS without certificate identity verification. This is a
+    copy of twisted.internet._sslverify.ClientTLSOptions with the identity
+    verification left out. For documentation, see the twisted documentation.
+    """
+
+    def __init__(self, hostname, ctx):
+        self._ctx = ctx
+        self._hostname = hostname
+        self._hostnameBytes = _idnaBytes(hostname)
```

Comment:
might be nice to copy `_idnaBytes` and `_tolerateErrors` (both of which seem to be trivial) rather than using the private impls from twisted.

Reply:
I agree it's not best practice to use private implementations. But, if I just copy _tolerateErrors, the pep8 test fails as it contains a bare "except:" statement. I'm not familiar enough with twisted / openssl to know which errors this try/except clause should catch. I can change this to "except Exception:" to both catch all errors and pass the pep8 test, but that doesn't seem right either. However it is the easiest solution. What do you think?

This is the _tolerateErrors code:

```
def _tolerateErrors(wrapped):
    def infoCallback(connection, where, ret):
        try:
            return wrapped(connection, where, ret)
        except:
            f = Failure()
            logger.exception("Error during info_callback")
            connection.get_app_data().failVerification(f)
    return infoCallback
```

Reply:
You can write `# noqa: E722` to disable the pep8 test. See https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/message.py#L583 for an example.

Reply:
That's an even easier solution. I'll add this comment.